pkgs/build-support/node/build-npm-package/default.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, fetchNpmDeps, callPackage, nodejs }:
+
+{ name ? "${args.pname}-${args.version}"
+, src ? null
+, srcs ? null
+, sourceRoot ? null
+, patches ? [ ]
+, npmDepsHash ? ""
+, nativeBuildInputs ? [ ]
+, buildInputs ? [ ]
+, ...
+} @ args:
+
+assert npmDepsHash != "";
+
+let
+  npmDeps = fetchNpmDeps {
+    inherit src srcs sourceRoot patches;
+    name = "${name}-npm-deps";
+    hash = npmDepsHash;
+  };
+
+  # TODO: how do we expose these like Rust? `npmHooks`?
```

Comment:
I think something like `npmPlatform` makes the most sense, that can have the hooks, `buildNpmPackage` and `fetchNpmDeps`.